cardshifter-server/src/main/java/com/cardshifter/server/model/MainServer.java

```diff
@@ -71,6 +71,11 @@ public MainServer(ServerConfiguration serverConfiguration) {
 	public Server start() {
         mods.loadExternal(mods.getDefaultModLocation());
         mods.loadExternal(Paths.get(config.getModsDirectory()));
+
+        if(mods.getAvailableMods().isEmpty()) {
+        	logger.warn("No mods were loaded from the different folders. Please check your installation.");
+        	throw new IllegalStateException("No mod loaded");
```

Comment:
But it's not a problem of the file system, the application is in illegal state because you want to load mods, but you did not provide any to load. I would argue that this is an illegal state for an application that need at least one mods so people can play, and as far as I know, we don't have a way to hot load a mod (which might be nice).  

I really like the custom exception though, might be a better idea.